### PR TITLE
Added cube and quad to builtins

### LINKED ISSUE #5137: Add cube and quad collada files to builtins
**Task (REQUIRED):**
Add cube.dae and quad.dae to builtins

**Expected outcome (REQUIRED):**
A cube and a quad model should be available from builtins

### PR DESCRIPTION
**Pull request by britzl**
Fixes #5137

### CODE REVIEW

**Comment by JCash** [comment]
I would also like a sphere.dae.

**Comment by aglitchman** [comment]
The cube and the quad size are 2 units (meters).

I'm not sure about other devs, but I usually like to use objects that have 1 unit in size (-0.5 to 0.5).

File: engine/engine/content/builtins/assets/meshes/quad.dae [hunk 1]
```diff
@@ -0,0 +1,98 @@
+<?xml version="1.0" encoding="utf-8"?>
+<COLLADA xmlns="http://www.collada.org/2005/11/COLLADASchema" version="1.4.1">
+  <asset><contributor><author></author><authoring_tool>FBX COLLADA exporter</authoring_tool><comments></comments></contributor><created>2012-10-10T12:34:45Z</created><keywords></keywords><modified>2012-10-10T12:34:45Z</modified><revision></revision><subject></subject><title></title><unit meter="1.000000" name="centimeter"></unit><up_axis>Y_UP</up_axis></asset>
```
Comment (JCash) [inline engine/engine/content/builtins/assets/meshes/quad.dae:3]:
```
<unit meter="1.000000" name="centimeter"
```

Is there a downside to having this model in centimeters?

**Comment by britzl** [comment]
> The cube and the quad size are 2 units (meters).
> 
> I'm not sure about other devs, but I usually like to use objects that have 1 unit in size (-0.5 to 0.5).

That's a good point! I took the files from one of our template projects.

File: engine/engine/content/builtins/assets/meshes/quad.dae (hunk 1 above)
Comment (britzl) [reply engine/engine/content/builtins/assets/meshes/quad.dae:3]: No idea! The file is from the template-basic-3d repo.

**Comment by dapetcu21** [comment]
> I'm not sure about other devs, but I usually like to use objects that have 1 unit in size (-0.5 to 0.5).

I actually disagree. The quad is useful for full screen filters, but there you want it to be in normalised device coordinates and for it to cover the whole screen it needs to be (-1, 1)

**Comment by aglitchman** [comment]
> I actually disagree. The quad is useful for full-screen filters, but there you want it to be in normalised device coordinates and for it to cover the whole screen it needs to be (-1, 1)

Yeah, maybe... 

I installed UE4, Unity, and Godot to see sizes of the standard objects. 
* UE4: cube is 100x100x100 uu (uu is cm by default), plane is 100x100 uu. So it's 1 meter in size.
* Godot: cube mesh is 2x2x2, quad mesh is 1x1.
* Unity: cube is 1x1x1, quad is 1x1.

**Comment by subsoap** [comment]
The one I usually use is -1 to 1 for full screen render target effects, but I've not tested -0.5 to 0.5 for that. Maybe the -1 to 1 can be included too and labeled rendertarget_quad?

**Comment by subsoap** [comment]
About the sizes of things, a standard does need to be set (or at least suggested) and communicated officially (like "Defold uses the metric system") for 3d space. Bulitin privatives should then reasonably conform to it. It would be useful to hear from @dragosha since he has been using 3d a lot recently.

**Comment by Dragosha** [comment]
> About the sizes of things, a standard does need to be set (or at least suggested) and communicated officially (like "Defold uses the metric system") for 3d space. Bulitin privatives should then reasonably conform to it. It would be useful to hear from @Dragosha since he has been using 3d a lot recently.

I'm not a big expert in this things. Just used a quad from examples. May be it'd cool to have several primitives with different size.

File: engine/engine/content/builtins/assets/meshes/quad.dae (hunk 1 above)
Comment (britzl) [reply engine/engine/content/builtins/assets/meshes/quad.dae]: Changed this to meters

**Comment by britzl** [comment]
* Added sphere (1 unit diameter)
* Changed quad to 1x1 unit
* Changed cube to 1x1x1 unit
* Added quad_2 which is 2x2 units

**Review by JCash — APPROVED** [review]

**Pull request merged**